modules/oe_subscriptions_anonymous/src/Plugin/EmailBuilder/AnonymousEmailBuilder.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_subscriptions_anonymous\Plugin\EmailBuilder;
+
+use Drupal\oe_subscriptions_anonymous\MailTemplate\MailTemplateHelper;
+use Drupal\symfony_mailer\Address;
+use Drupal\symfony_mailer\EmailFactoryInterface;
+use Drupal\symfony_mailer\EmailInterface;
+use Drupal\symfony_mailer\Processor\EmailBuilderBase;
+
+/**
+ * Defines the Email Builder plug-in for oe_subscriptions_anonymous module.
+ *
+ * @EmailBuilder(
+ *   id = "oe_subscriptions_anonymous",
+ *   label = @Translation("Anonymous subscription"),
+ *   sub_types = {
+ *     "subscription_create" = @Translation("Subscription creation"),
+ *     "user_subscriptions_access" = @Translation("Subscriptions access request"),
+ *   },
+ * )
+ */
+class AnonymousEmailBuilder extends EmailBuilderBase {
+
+  /**
+   * Creates the parameters for a newly created email.
+   *
+   * @param \Drupal\symfony_mailer\EmailInterface $email
+   *   The email to modify.
+   * @param array $params
+   *   The email parameters.
+   */
+  public function createParams(EmailInterface $email, array $params = []): void {
+    $mail_template = MailTemplateHelper::getKeyClass($email->getSubType());
+
+    foreach ($mail_template::getParameters() as $param) {
+      assert($params[$param] != NULL);
+      $email->setParam($param, $params[$param]);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function fromArray(EmailFactoryInterface $factory, array $message): EmailInterface {
+    return $factory->newTypedEmail($message['module'], $message['key'], $message['params']);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(EmailInterface $email): void {
+    $params = [];
+    $mail_template = MailTemplateHelper::getMailTemplate($email->getSubType());
+
+    foreach ($mail_template->getParameters() as $param) {
+      $params[$param] = $email->getParam($param);
```

Comment:
We need to have these defined as variables so that they can be used in template overrides.
We also need to define the mailer_override configurations so that people can easily override the text.

Reply:
Added variables and configurations to override